internal/pkg/manifest/rd_web_svc.go

```diff
@@ -86,6 +86,11 @@ type RequestDrivenWebServiceHttpConfig struct {
 	Private                  Union[bool, VPCEndpoint] `yaml:"private"`
 }
 
+// PrivateEnabled returns true if r is configured for private ingress.
+func (r *RequestDrivenWebServiceHttpConfig) PrivateEnabled() bool {
+	return r.Private.Basic || !r.Private.Advanced.IsZero()
```

Comment:
ohh unrelated to this PR - does having `bool` for `r.Private.Basic` play nicely with environment override?
```
http:
   private: true

environments:
  test:
     # did not specify http.private
```
What is the result of this manifest for `test` env's `http.private`?

Reply:
great question! this was a rabbit hole for me :) 

first issue: i forgot to add the new union type (`[bool, VPCEndpoint]`) to the list of transformers. (done)

second issue: i assumed that mergo would merge the bool values correctly, but then i started writing tests and found https://github.com/imdario/mergo/issues/89. Apparently mergo doesn't override dst with src if it's a zero value. (you already figured this out, with the `basicTransformer`😊 - but it only works for pointers) i fixed this for the union type by checking if the Union type thinks the value was set. Some examples of how it works (now!) - there's also some tests in `loader_test.go`: 
```
# 1 - will override
environments:
  test:
    http:
      private: true
# 2 - will override
environments:
  test:
    http:
      private: false
# 3 - will not override
environments:
  test:
    http:
      alias: "example.com"
# 4 - will not override
environments:
  test:
    cpu: 1024
```